ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/functions/useRows.ts

```diff
@@ -44,7 +44,11 @@ export function useRows({ archived }: { archived: boolean }): Output {
 
   // Append new function data.
   useEffect(() => {
-    if (functionsRes.data && functionsRes.data.page.page > functionsData.lastPage) {
+    if (!functionsRes.data || functionsRes.isLoading) {
+      return;
+    }
+
+    if (functionsRes.data.page.page > functionsData.lastPage) {
```

Comment:
How does this know to `setFunctionsData` on only archived vs. not archived?

Reply:
The data is cleared out when `archived` changes:
https://github.com/inngest/inngest/blob/main/ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/%5BenvironmentSlug%5D/functions/useRows.ts#L33

This whole hook needs to be refactored. It's probably the worst code I've written here